packages/ra-ui-materialui/src/field/TranslatableFieldsTabContent.tsx

```diff
@@ -57,28 +77,25 @@ export const TranslatableFieldsTabContent = (
         >
             {Children.map(children, field =>
                 field && isValidElement<any>(field) ? (
-                    <ResourceContextProvider
-                        value={resource}
-                        key={field.props.source}
-                    >
-                        <RecordContextProvider value={record}>
-                            <SourceContextProvider value={sourceContext}>
-                                <div>
-                                    {addLabel ? (
-                                        <Labeled
-                                            resource={resource}
-                                            label={field.props.label}
-                                            source={field.props.source}
-                                        >
-                                            {field}
-                                        </Labeled>
-                                    ) : (
-                                        field
-                                    )}
-                                </div>
-                            </SourceContextProvider>
-                        </RecordContextProvider>
-                    </ResourceContextProvider>
+                    <RecordContextProvider value={recordForLocale}>
```

Comment:
missing key. Also, it's the same record for all the fields, why do we create the RecordContext inside the Children.map? It should be outside.